xarray/tests/test_missing.py

```diff
@@ -255,19 +255,30 @@ def test_interpolate():
     assert_equal(actual, expected)
 
 
-def test_interpolate_nonans():
-
-    vals = np.array([1, 2, 3, 4, 5, 6], dtype=np.float64)
-    expected = xr.DataArray(vals, dims="x")
-    actual = expected.interpolate_na(dim="x")
-    assert_equal(actual, expected)
-
-
 @requires_scipy
-def test_interpolate_allnans():
-    vals = np.full(6, np.nan, dtype=np.float64)
+@pytest.mark.parametrize(
+    "method,vals",
+    [
+        pytest.param(method, vals, id=f"{desc}:{method}")
+        for method in [
+            "linear",
+            "nearest",
+            "zero",
+            "slinear",
+            "quadratic",
+            "cubic",
+            "polynomial",
+        ]
```

Comment:
Would be nice if this wasn't hardcoded. But they seem hidden inside a function currently:
https://github.com/pydata/xarray/blob/5b322c9ea18f560e35857edcb78efe4e4f323551/xarray/core/missing.py#L471-L486

Reply:
thanks @Illviljan ! yeah I considered trying to import these but opted not to modify xarray/core. but if you think creating a global variable or something is preferable I'm happy to add that to the pr!

Reply:
I think we'll skip this for now. We'll add it when it causes problems.